nx/lib/nx/defn/kernel.ex

```diff
@@ -233,6 +233,13 @@ defmodule Nx.Defn.Kernel do
     grad
   end
 
+  @doc """
+  Vectorizes `fun` over given `args`.
```

Comment:
Can we add some examples and more explanation? :) I am struggling to wrap my head around the current API. I assume that `fun` receives the vectorized tensor? Are there any restrictions on `args`? What are the possible values for `in_axes`?

Reply:
Perhaps handling `dot` is a good example of how the code will change before and after with and without vmap. :)